### PR TITLE
Feature/schema middleware

### PR DESCRIPTION
**Pull request by wgermancutraro**
## Summary

I created a "schemas" folder inside the validations folde.r There the idea is to have all schemas that exists in the application. Then in the "middlewares" folder, exists a validatorSchema.js file that assurance the validations provided.

## Known Issues

-

## Trello Card

https://trello.com/c/mPCmw4dw/27-middlewares

### CODE REVIEW

File: app/mappers/userMapper.js [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
-    this.first_name = name;
-    this.last_name = last_name;
+    this.first_name = firstName;
+    this.last_name = lastName;
     this.email = email;
     this.created_at = createdAt;
-    this.updated_at = createdAt;
+    this.updated_at = updatedAt;
   }
 }
```
Comment (martinremis) [inline app/mappers/userMapper.js]: I did not notice this before. Why is this a class?

File: app/middlewares/validateSchema.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+const validateSchema = schema => [
+  checkSchema(schema),
+  (req, _, next) => {
+    const errors = validationResult(req);
+    if (errors.isEmpty()) next();
+    else {
+      const errorMessage = errors.array()[0].msg;
+
+      logger.error(`Validation error: ${req.path}: ${errorMessage}`);
+      next(validationError(errorMessage));
```
Comment (martinremis) [inline app/middlewares/validateSchema.js]: Here you are just throwing the first error. Could you response all the validation errors or there is some limitation and/or requirement to not do so?

File: app/constants/errorMessages.js [hunk 3]
```diff
@@ -3,8 +3,9 @@ const errorMessages = {
   invalidEmail: 'Email no válido',
   invalidEmailResource: 'Email no válido, no pertenece a Wolox',
   invalidPassword: 'Contraseña no válida',
+  invalidPasswordLength: 'La contraseña debe tener como minimo 8 caracteres',
   passwordRequired: 'Contraseña requerida',
-  nameRequired: 'Nombre requerido',
+  firstNameRequired: 'Nombre requerido',
   lastNameRequired: 'Apellido requerido',
   userAlreadyExist: 'Usuario existente'
 };
```
Comment (martinremis) [inline app/constants/errorMessages.js:11]: Response and error messages are usually in English. No need to change it anyway, they are just string messages :smile:

**Review by martinremis — CHANGES_REQUESTED** [review]

File: app/mappers/userMapper.js (hunk 1 above)
Comment (wgermancutraro) [reply app/mappers/userMapper.js]: That class is to map the response (like a serializer) i created that in order to response to the front-end formatted fields. I pushed a refactor version.

File: app/middlewares/validateSchema.js (hunk 2 above)
Comment (wgermancutraro) [reply app/middlewares/validateSchema.js]: I did that to make it more easy to catch. In muy personal opinion, in most cases is easier to the front-end to receive one error. But i change it in order to receive all errors.

**Review by martinremis — APPROVED** [review]

**Pull request merged**